piwik-sdk/src/main/java/org/piwik/sdk/DownloadTracker.java

```diff
@@ -50,20 +68,26 @@ public DownloadTracker(Tracker tracker, TrackMe baseTrackMe) {
         mContext = piwik.getContext();
         mPackageName = mContext.getPackageName();
         mPackMan = mContext.getPackageManager();
+        try {
+            mPkgInfo = mPackMan.getPackageInfo(mPackageName, 0);
+            mName = mPkgInfo.packageName;
+            mVersion = getVersion();
+            if (mVersion == null){
+                throw new PackageManager.NameNotFoundException("Version could not be found");
+            }
+        }
+        catch (PackageManager.NameNotFoundException e) {
+            e.printStackTrace();
+        }
     }
 
     public void trackOnce(@NonNull Extra extra) {
-        try {
-            PackageInfo pkgInfo = mContext.getPackageManager().getPackageInfo(mContext.getPackageName(), 0);
-            String firedKey = "downloaded:" + pkgInfo.packageName + ":" + pkgInfo.versionCode;
-            synchronized (TRACK_ONCE_LOCK) {
-                if (!mPreferences.getBoolean(firedKey, false)) {
-                    mPreferences.edit().putBoolean(firedKey, true).apply();
-                    trackNewAppDownload(extra);
-                }
+        String firedKey = "downloaded:" + mName + ":" + mVersion;
```

Comment:
The idea is to use getVersion() through out the code instead of mVersion, which load either provide a manually set version or pull one from the context.